File: HashMap-Dictionary/TwoSum.java
import java.util.*;
class TwoSum{
    public static int[] twoSum(int[] nums, int target){
        Map<Integer, Integer> complementMap = new HashMap<>();

        for(int i = 0; i<nums.length; i++){
            int currentNum = nums[i];
            int complement = target - currentNum;

            if(complementMap.containsKey(complement)){
                return new int[]{complementMap.get(complement), i};
            }

            complementMap.put(currentNum, i);
        }

        return new int[]{};
    }

    public static void main(String[] args) {
        int[] nums = {2,7,11,15};
        int target = 22;
        int[] result = twoSum(nums, target);
        System.out.println("[" + result[0] + ", "+ result[1]+ "]");
    }
}

File: HashMap-Dictionary/TwoSum.py
class TWOSUM:
    def two_sum(self, nums, target):
        complement_dict = {}

        for i, currentNum in enumerate(nums): # idiomatic line
            
            complement = target - currentNum

            if complement in complement_dict:
                return [complement_dict[complement],i]
            
            complement_dict[currentNum] = i
        return []
if __name__ == "__main__":    
    nums = [2, 7, 11, 15]
    target = 9
    twosum = TWOSUM()
    result = twosum.two_sum(nums, target)
    print(result)